### PR TITLE
[Testing] Cleanup phan file and make explicit the settings we are using

### PR DESCRIPTION
**Pull request by johnsaigle**
## Brief summary of changes

* Explicitly declare severity level (unchanged from before)
* ~~Explicitly declare targeted version of PHP version~~ This could cause a problem with Travis
* Update link from the etsy repo to phan
* Remove a duplicate, conflicting setting
* ~~Remove backward compatibility check because we don't need it~~ Dave wants this to stay
* Change tabs to spaces

### CODE REVIEW

File: .phan/config.php [hunk 1]
```diff
@@ -1,20 +1,33 @@
 <?php
 
 return [
-	"backward_compatibility_checks" => true,
```
Comment (johnsaigle) [inline .phan/config.php]: We're firmly on PHP 7 now so I don't think we need to check this.

Comment (driusan) [reply .phan/config.php]: I believe this enables backwards compatibility checks against target_php_version, not against PHP7.

File: .phan/config.php [hunk 2]
```diff
@@ -1,20 +1,38 @@
...
+    // Toggling the below rule to true is the suggested fix by the phan devs.
+    // see https://github.com/phan/phan/issues/1650.
+    //
+    // When the module is refactored, this line should be deleted.
+	"ignore_undeclared_variables_in_global_scope" => true,
+    // FIXME: We should add this.
+    "dead_code_detection" => false,
+    // FIXME: We should add this. Note that dead_code_detection also covers
+    // unused_variable_detection. It might be a good idea to enable this first.
+    "unused_variable_detection" => false,
```
Comment (driusan) [inline .phan/config.php]: Are these finding things? phpmd already did/does these things, so we should have 0 results found.

Comment (johnsaigle) [reply .phan/config.php]: It's finding many things... I enabled just the `unused_variable_detection` rule and got 247 results after running phan.

PHPMD is also only configured to run on `php/libraries/`. 72 of the above results occurred in that folder so PHPMD is definitely missing something OR phan is giving a ton of false positives. Taking a look at the results though they seem accurate.

File: .phan/config.php (hunk 1 above)
Comment (johnsaigle) [reply .phan/config.php]: The references I can find seem to specify PHP 5 --> PHP 7 changes. I don't think it has to do with target_php_version

https://github.com/phan/phan/wiki/Tutorial-for-Analyzing-a-Large-Sloppy-Code-Base#slowly-ramp-up-the-strength-of-the-analysis

https://github.com/phan/phan/wiki/Phan-Config-Settings#backward_compatibility_checks

File: .phan/config.php (hunk 2 above)
Comment (johnsaigle) [reply .phan/config.php]: (Some of the above errors are already fixed by my PRs that add new Phan rules.)

File: .phan/config.php (hunk 1 above)
Comment (PapillonMcGill) [reply .phan/config.php]: from the 2nd link:
```
Backwards Compatibility Checking. This is slow and expensive, but you should consider running it before upgrading your version of PHP to a new version that has backward compatibility breaks
```

Reading this, it is now specified for php 7 or 5.  We might have php8 in the future when thing break with php7.  Even if this possibility is low.  
But as @johnsaigle mentionned, all the code is done for php7 and the cost of this is high. WIll help with time to check.

Comment (driusan) [reply .phan/config.php]: Even if it's "high cost", high is at most on the order of seconds for phan, in a CI run that takes over 20 minutes to run the integration tests. I think we should still keep it.

**Comment by xlecours** [comment]
@johnsaigle could you fix indentation?

**Comment by johnsaigle** [comment]
what's wrong with the indentation?

**Comment by xlecours** [comment]
It's a mix of tabs and spaces

**Comment by PapillonMcGill** [comment]
@johnsaigle @xlecours ask you to replace tab with spaces

**Comment by johnsaigle** [comment]
that's no good! will do.

**Comment by johnsaigle** [comment]
I got pattern not found `^I` when running your command. Are you sure the formatting is bad? It may just be the way GitHub is displaying the diff.

File: .phan/config.php [hunk 3]
```diff
@@ -1,63 +1,79 @@
...
+        // This is excluded because the API is in the process of being converted
+        // to a proper LORIS module. Everything in this folder will be deleted
+        // so we're not going to spend time cleaning it up.
+        "htdocs/api/",
+    ],
+    "autoload_internal_extension_signatures" => [
         // Xdebug stubs are bundled with Phan 0.10.1+/0.8.9+ for usage,
         // because Phan disables xdebug by default.
-        'xdebug'     => 'vendor/phan/phan/.phan/internal_stubs/xdebug.phan_php',
+        "xdebug"     => "vendor/phan/phan/.phan/internal_stubs/xdebug.phan_php",
```
Comment (PapillonMcGill) [inline .phan/config.php]:
```suggestion
        "xdebug" => "vendor/phan/phan/.phan/internal_stubs/xdebug.phan_php",
```

**Comment by driusan** [comment]
`^I` means ctrl-I (or tab), not caret I. You may need to use `ctrl-m ctrl-I`

**Comment by johnsaigle** [comment]
Yep I know it means Ctrl, I needed to be in a specific Vim mode. Xavier and I cleared it up via Slack.

The file is fixed now.

**Comment by xlecours** [comment]
@johnsaigle the file is still all messed up.

**Comment by johnsaigle** [comment]
@xlecours Hmmm I thought I had fixed it. How is it now?

**Comment by xlecours** [comment]
> @xlecours Hmmm I thought I had fixed it. How is it now?

Almost there. Line#4 is still offending.

File: .phan/config.php [hunk 4]
```diff
@@ -1,54 +1,25 @@
...
-		"vendor",
-		"htdocs/api/",
-	],
-    'autoload_internal_extension_signatures' => [
-        // Xdebug stubs are bundled with Phan 0.10.1+/0.8.9+ for usage,
-        // because Phan disables xdebug by default.
-        'xdebug'     => 'vendor/phan/phan/.phan/internal_stubs/xdebug.phan_php',
-    ],
-
+"backward_compatibility_checks" => true,
```
Comment (xlecours) [inline .phan/config.php]:
```suggestion
    "backward_compatibility_checks" => true,
```

**Comment by johnsaigle** [comment]
@xlecours thanks!

**Comment by xlecours** [comment]
@johnsaigle Still need to be rebased.

**Comment by johnsaigle** [comment]
😭

**Comment by johnsaigle** [comment]
I'll fix it

**Comment by johnsaigle** [comment]
@xlecours Let me know if my fix works.

**Comment by xlecours** [comment]
> > @xlecours Hmmm I thought I had fixed it. How is it now?
> 
> Almost there. Line#4 is still offending.

bis

**Comment by johnsaigle** [comment]
How's that?

**Comment by johnsaigle** [comment]
Many of the Travis errors are caused by phan analyzing the `htdocs/api` directory, which is deleted on 22.0-release. When that branch is merged into `master` this shouldn't be a problem.

File: .phan/config.php [hunk 5]
```diff
@@ -1,70 +1,87 @@
...
+    // This is false on the assumption that that's a typo.
+    // It doesn't seem to have any effect on LORIS's codebase anyways.
+    "quick_mode" => false,
+    "analyze_signature_compatibility" => true,
+    // The default severity level for phan is 5. After removing all the
+    // suppressed rules, we should consider reducing this value to detect more
+    // suspicious code.
+    "minimum_severity" => 5,
+    // TODO: Update this value to the minimum required by LORIS.
+    "target_php_version" => "7.3",
```
Comment (driusan) [inline .phan/config.php]: This should not be explicitly set, or Travis won't be testing the right version.

File: .phan/config.php [hunk 6]
```diff
@@ -1,70 +1,87 @@
...
         "PhanTypeArraySuspiciousNullable",
-        "PhanNoopNew"
-	],
-	"analyzed_file_extensions" => ["php", "inc"],
-	"directory_list" => [
-		/* This doesn't include php/installer, because there's
-		   (intentionally) classes in the installer namespace
+        "PhanNoopNew",
+        "PhanNonClassMethodCall",
+        "PhanTypeSuspiciousStringExpression"
```
Comment (driusan) [inline .phan/config.php]: You seem to have unnecessarily added some new exception types, probably from a bad rebase..

File: .phan/config.php [hunk 7]
```diff
@@ -1,70 +1,87 @@
...
+    ],
+    "analyzed_file_extensions" => ["php", "inc"],
+    "directory_list" => [
+        // This doesn't include php/installer, because there's
+        //  (intentionally) classes in the installer namespace
```
Comment (driusan) [inline .phan/config.php]: Unless I'm misreading this diff these lines are now duplicated

**Comment by ridz1208** [comment]
@driusan are all your comments addressed ?

@johnsaigle conflict

**Comment by driusan** [comment]
No

**Comment by johnsaigle** [comment]
@driusan I think I've now incorporated all of your suggestions

File: .phan/config.php [hunk 8]
```diff
@@ -1,66 +1,74 @@
...
         "PhanTypePossiblyInvalidDimOffset",
-		"PhanUndeclaredMethod",
+        "PhanUndeclaredMethod",
         "PhanTypeMismatchDimFetch",
-		"PhanTypeMismatchReturn",
-		"PhanUndeclaredClassMethod",
-		"PhanTypeMismatchArgument",
-		"PhanTypeMismatchProperty",
+        "PhanTypeMismatchReturn",
+        "PhanUndeclaredVariableDim",
```
Comment (driusan) [inline .phan/config.php]: When I look at this with `?w=1`, this seems to be accidentally sneaking PhanUndeclaredVariableDim into the list of ignores.

**Pull request merged**